src/runtime/cuda/cuda_hashtable.cu

```diff
@@ -4,11 +4,11 @@
  * \brief Device level functions for within cuda kernels.
  */
 
-#include <cub/cub.cuh>
 #include <cassert>
 
 #include "cuda_hashtable.cuh"
 #include "../../kernel/cuda/atomic.cuh"
```

Comment:
The `src/kernel` folder would be deprecated, please include `src/array/cuda/atomic.cuh` instead.

Reply:
I will remove the legacy codes and fix this include later.